### DISCUSSION TITLE
Modify tools prompt to make sure the <tools> XML is surrounded by ```xml

### DISCUSSION THREAD

**Discussion by ogomezba** [olimorris/codecompanion.nvim#416] [Ideas]
Hi! First of all, thanks for the amazing job with the plugin. I've been playing with it a bit and it provides great help and an amazing workflow.

I've noticed that, while using the @editor tool with Anthropic (model `claude-3-5-haiku-20241022`), I've been experiencing some problems with the `<tools>` XML generated. It was being generated quite consistently **without** the ```xml markdown syntax, just the plain XML in the chat buffer. This was preventing the parser from recognizing the tool XML, so it could not launch the corresponding action.

By including this last sentence to the **Key Points** prompt section of the `@editor` tool, situation has improved remarkably:

```
3. **Key Points**:
  - **Only use when prompted** by user (e.g., "can you update the code?", "update the buffer...")
  - Ensure XML is **valid and follows the schema**
  - **Include indentation** in your code
  - **Don't escape** special characters
  - **Wrap code in a CDATA block**, the code could contain characters reserved by XML
  - **Make sure the tools xml block is surrounded by ```xml
 ```

I'm no expert at all in LLM prompting so I'm unaware of the impact of such change but thought it could be useful to share as others might have had the same experience.

**Comment by olimorris** [comment]
Thanks for your feedback. This seems like a great suggestion and please make a pull request for this.

**Comment by ogomezba** [comment]
Implemented in https://github.com/olimorris/codecompanion.nvim/pull/417
